MoltenVK/MoltenVK/GPUObjects/MVKPipeline.mm

```diff
@@ -507,6 +507,16 @@
 	_outputControlPointCount = reflectData.numControlPoints;
 	mvkSetOrClear(&_tessInfo, _isTessellationPipeline ? pCreateInfo->pTessellationState : nullptr);
 
+	// Handles depth attachment being used as input attachment. However, it does not solve the issue when
+	// the pipeline is created without render pass (dynamic rendering) since we won't be able to know
+	// which resources will be used when rendering. Needs to be done before we do shaders
+	// Potential solution would be to generate 2 pipelines, one with the workaround for the Metal issue
+	// and one without it, and decide at bind time once we know the resources which one to use.
```

Comment:
I will take a look to move the RP case to `MVKRenderSubpass::populatePipelineRenderingCreateInfo()`.

Regarding the dynamic rendering case, it does state the format of the depth/stencil, but I don't think there's a way to know the format of the used input textures at pipeline creation time since we only have the descriptor set layout. So we wouldn't know if an input texture is the same as the depth/stencil until bind time. I'll give it a thought though